### ISSUE TITLE
set-rsa-pass will zero the private key if provided an invalid password

### ISSUE THREAD

**Issue by hmanning77** [OpenVPN/easy-rsa#204]
The `set_pass` function passes the private key filename as both the `-in` and `-out` arguments. This works when the command is successful, but if the user types provides an invalid new password (or mistypes it when asked to repeat it) OpenSSL will abort leaving the key file empty. Example output below:

```
hugh@anastacia ~/S/P/e/t/easyrsa3> ./easyrsa init-pki

init-pki complete; you may now create a CA or requests.
Your newly created PKI dir is: /home/hugh/Software/Projects/easy-rsa/test/easyrsa3/pki

hugh@anastacia ~/S/P/e/t/easyrsa3> ./easyrsa build-ca
Enter New CA Key Passphrase: 
Re-Enter New CA Key Passphrase: 
Generating RSA private key, 2048 bit long modulus
................................+++
.......................................................+++
e is 65537 (0x010001)
You are about to be asked to enter information that will be incorporated
into your certificate request.
What you are about to enter is what is called a Distinguished Name or a DN.
There are quite a few fields but you can leave some blank
For some fields there will be a default value,
If you enter '.', the field will be left blank.
-----
Common Name (eg: your user, host, or server name) [Easy-RSA CA]:

CA creation complete and you may now import and sign cert requests.
Your new CA certificate file for publishing is at:
/home/hugh/Software/Projects/easy-rsa/test/easyrsa3/pki/ca.crt

hugh@anastacia ~/S/P/e/t/easyrsa3> ./easyrsa build-client-full test
Generating a 2048 bit RSA private key
.............................................................................+++
......+++
writing new private key to '/home/hugh/Software/Projects/easy-rsa/test/easyrsa3/pki/private/test.key.pJ0f71I0QN'
Enter PEM pass phrase:
Verifying - Enter PEM pass phrase:
-----
Using configuration from ./openssl-easyrsa.cnf
Enter pass phrase for /home/hugh/Software/Projects/easy-rsa/test/easyrsa3/pki/private/ca.key:
Can't open /home/hugh/Software/Projects/easy-rsa/test/easyrsa3/pki/index.txt.attr for reading, No such file or directory
140674741059520:error:02001002:system library:fopen:No such file or directory:crypto/bio/bss_file.c:74:fopen('/home/hugh/Software/Projects/easy-rsa/test/easyrsa3/pki/index.txt.attr','r')
140674741059520:error:2006D080:BIO routines:BIO_new_file:no such file:crypto/bio/bss_file.c:81:
Check that the request matches the signature
Signature ok
The Subject's Distinguished Name is as follows
commonName            :ASN.1 12:'test'
Certificate is to be certified until Apr  5 05:19:06 2028 GMT (3650 days)

Write out database with 1 new entries
Data Base Updated
hugh@anastacia ~/S/P/e/t/easyrsa3> ls -l pki/private
total 8
-rw------- 1 hugh hugh 1766 Apr  8 17:18 ca.key
-rw------- 1 hugh hugh 1854 Apr  8 17:19 test.key
hugh@anastacia ~/S/P/e/t/easyrsa3> ./easyrsa set-rsa-pass test

If the key is currently encrypted you must supply the decryption passphrase.
You will then enter a new PEM passphrase for this key.

Enter pass phrase for /home/hugh/Software/Projects/easy-rsa/test/easyrsa3/pki/private/test.key:
writing RSA key
Enter PEM pass phrase:
unable to write key
139988222488512:error:28069065:UI routines:UI_set_result:result too small:crypto/ui/ui_lib.c:765:You must type in 4 to 1024 characters
139988222488512:error:2807106B:UI routines:UI_process:processing error:crypto/ui/ui_lib.c:493:while reading strings
139988222488512:error:0906406D:PEM routines:PEM_def_callback:problems getting password:crypto/pem/pem_lib.c:64:
139988222488512:error:0906906F:PEM routines:PEM_ASN1_write_bio:read key:crypto/pem/pem_lib.c:336:

Easy-RSA error:

Failed to change the private key passphrase. See above for possible openssl
error messages.
hugh@anastacia ~/S/P/e/t/easyrsa3 [1]> ls -l pki/private
total 4
-rw------- 1 hugh hugh 1766 Apr  8 17:18 ca.key
-rw------- 1 hugh hugh    0 Apr  8 17:19 test.key
hugh@anastacia ~/S/P/e/t/easyrsa3> 

```

**Comment by TinCanTech** [comment]
This is a bug in easyrsa.

OpenSSL man page states:
"The output filename should **not** be the same as the input filename."

**Comment by luizluca** [comment]
I have already noticed that some months ago at https://github.com/OpenVPN/easy-rsa/pull/213:

My current fix for that is:

https://github.com/luizluca/easy-rsa/tree/fix_setpass

However, it is based on other non-merged changes, specially the one that maintains tempfiles as a list and not in different variables. I have already a bunch of PR opens to process before pushing more stuff.

**Comment by TinCanTech** [comment]
Closed in #276
